src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -515,110 +452,169 @@ public void updateSubscribers()
+        if (isCitizensDirty || hasNewSubscribers)
+        {
+            for (CitizenData citizen : citizens.values())
             {
-                subscribers
-                        .stream()
-                        .filter(player -> permissions.isDirty() || !oldSubscribers.contains(player)).forEach(player -> {
-                    Permissions.Rank rank = getPermissions().getRank(player);
-                    MineColonies.getNetwork().sendTo(new PermissionsMessage.View(this, rank), player);
-                });
+                if (citizen.isDirty() || hasNewSubscribers)
+                {
+                    ColonyViewCitizenViewMessage msg = new ColonyViewCitizenViewMessage(this, citizen);
+
+                    subscribers.stream()
+                               .filter(player -> citizen.isDirty() || !oldSubscribers.contains(player))
+                               .forEach(player -> MineColonies.getNetwork().sendTo(msg, player));
+                }
             }
+        }
+    }
 
-            //  Citizens
-            if (isCitizensDirty || hasNewSubscribers)
+    private void sendPermissionsPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasNewSubscribers)
+    {
+        if(permissions.isDirty() || hasNewSubscribers)
+        {
+            subscribers
+                    .stream()
+                    .filter(player -> permissions.isDirty() || !oldSubscribers.contains(player)).forEach(player -> {
+                Permissions.Rank rank = getPermissions().getRank(player);
+                MineColonies.getNetwork().sendTo(new PermissionsMessage.View(this, rank), player);
+            });
+        }
+    }
+
+    private void sendColonyViewPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasNewSubscribers)
+    {
+        if (isDirty || hasNewSubscribers)
+        {
+            for (EntityPlayerMP player : subscribers)
             {
-                for (CitizenData citizen : citizens.values())
+                boolean isNewSubscriber = !oldSubscribers.contains(player);
+                if (isDirty || isNewSubscriber)
                 {
-                    if (citizen.isDirty() || hasNewSubscribers)
-                    {
-                        ColonyViewCitizenViewMessage msg = new ColonyViewCitizenViewMessage(this, citizen);
-
-                        subscribers.stream()
-                                   .filter(player -> citizen.isDirty() || !oldSubscribers.contains(player))
-                                   .forEach(player -> MineColonies.getNetwork().sendTo(msg, player));
-                    }
+                    MineColonies.getNetwork().sendTo(new ColonyViewMessage(this, isNewSubscriber), player);
                 }
             }
+        }
+    }
 
-            //  Buildings
-            if (isBuildingsDirty || hasNewSubscribers)
+    private static boolean hasNewSubscribers(Set<EntityPlayerMP> oldSubscribers, Set<EntityPlayerMP> subscribers)
+    {
+        for (EntityPlayerMP player : subscribers)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'player' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)